eeps/eep-0064.md

```diff
@@ -0,0 +1,491 @@
+verbatim strings, though.  So you have to choose between either
+verbatim with ending newline or having to escape backslashes.
+
+A slightly annoying consequence of verbatim strings with a fixed
+end marker there is no possibility to create a string containing
+such an end marker.  Therefore this EEP, as a possible extension,
+suggests allowing not only triple-quoted strings,
+but 3-or-more-quoted strings.  Then a start and end marker
+can be chosen that is not part of the string,
+and any string can be created.  This is a not so pretty solution
+for a very rare corner case.
+
+### Triple-Quoted String Scanner Token
+
+A triple-quoted string must be a token that the scanner recognizes
+as a string, which makes it suitable for a documentation attribute
+value term.  It starts and ends with three double quotes: `"""`.
+
+Double quotes, `"`, are chosen because normal Erlang strings use them
+and this is just a new variant.  Since double quotes are used
+a triple-quoted string shall, as a normal string,
+produce a list of code-points.
+
+It would be more convenient if a triple-quoted string produced
+an UTF-8 binary, but that would be a surprising feature for double quotes,
+and the documentation build process can work around this by converting
+the code-point list (`string()`) into the needed binary chunk.
+
+In source code a triple-quoted string is valid within a binary,
+so producing a Unicode binary is reasonably straightforward:
+
+    X = <<"""
+        Line 1
+        Line 2
+        """/utf8>>
+
+The 2 + 7 characters overhead is not exhausting since we are targeting
+multi-line strings here.
+
+As a future expansion it has been proposed to use [Sigils][] (prefixes)
+for specialized strings such as regular expressions,
+interpolated variables ([PR-7343][]), Unicode binary strings, etc.
+For example: `X = ~u"Tschüß"` for an UTF-8 encoded string.
+
+### Triple-Quoted String Start
+
+After the starting `"""` only white-space is allowed
+up to the end of the line.
+
+As a possible future expansion we might allow text here
+that shouldn't be part of the string content, but could be
+a hint for e.g. syntax highlighting and indentation handling
+in an editor / pretty printer.
+
+    -doc """ md
+        Markdown content
+        * Bullet list
+        """.
+
+The scanner does not need to have any special treatment
+for the string content on the line after the starting
+`"""` except that it should not search for an ending `"""`.
+
+A later step, preferably done by the parser, strips the
+characters up to and including the newline from the string content.
+
+If any of these characters is not white-space,
+the parser reports a syntax error.
+
+### Triple-Quoted String End
+
+All characters are collected as they are (verbatim) and becomes
+the string content.
+
+A triple-quoted string ends with newline followed by optional white-space
+and then `"""`.  This completes the scanner token.
+
+A later step, preferably done by the parser, uses the white-space
```

Comment:
Is it really preferable to strip indentation in the parser? I can't see why this shouldn't be done in the lexer. If the lexer strips indentation, the token appears with the actual string content in the sequence of tokens, as with normal strings where `erl_scan` handles the escape sequences.

Reply:
That would be convenient, we maybe wouldn't even need a new token, but having the lexer/scanner strip indentation would mean an extra pass over the string's content, which is something the scanner doesn't do today, so it would complicate the scanner structure.

Furthermore, I suspect that to have a source code preserving compiler (robust parser), which is on our ToDo list, also skipped characters should be passed through the parser to the abstract format.

But there might be no actual strong reasons against doing it in the scanner.  We can start there and see what happens...